### PR TITLE
Use DOTNET_JITAllowOptionalRelocs=0 on Linux/mac

### PR DESCRIPTION
**Pull request by EgorBo**

### CODE REVIEW

File: src/jit-dasm-pmi/Program.cs [hunk 1]
```diff
@@ -387,6 +388,13 @@ void AppendEnvironmentVariableToPmiEnv(string varName, string varValue)
                     }
                 }
 
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    // Disable fragile relocs on non-Windows platforms, see
+                    // https://github.com/dotnet/runtime/issues/87842
+                    AddEnvironmentVariable("DOTNET_JITAllowOptionalRelocs", "0");
```
Comment (jakobbotsch) [inline src/jit-dasm-pmi/Program.cs:395]: Environment variables are case sensitive, so this should first be renamed in dotnet/runtime according to @BruceForstall's feedback.

Comment (EgorBo) [reply src/jit-dasm-pmi/Program.cs:395]: > Environment variables are case sensitive, so this should first be renamed in dotnet/runtime according to @BruceForstall's feedback.

Are they actually? e.g. `dotnet_jitdisasm=` seems to be working fine to me

Comment (jakobbotsch) [reply src/jit-dasm-pmi/Program.cs:395]: On Unixes they are

Comment (EgorBo) [reply src/jit-dasm-pmi/Program.cs:395]: ugh...

File: src/jit-dasm/Program.cs [hunk 2]
```diff
@@ -344,6 +345,13 @@ public int GenerateAsm(List<AssemblyInfo> assemblyInfoList)
                     AddEnvironmentVariable("DOTNET_AltJitName", altJit);
                 }
 
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    // Disable fragile relocs on non-Windows platforms, see
+                    // https://github.com/dotnet/runtime/issues/87842
+                    AddEnvironmentVariable("DOTNET_JITAllowOptionalRelocs", "0");
```
Comment (MichalPetryka) [inline src/jit-dasm/Program.cs:352]:
```suggestion
                    AddEnvironmentVariable("DOTNET_JitEnableOptionalRelocs", "0");
```

File: src/jit-dasm-pmi/Program.cs (hunk 1 above)
Comment (MichalPetryka) [inline src/jit-dasm-pmi/Program.cs:395]:
```suggestion
                    AddEnvironmentVariable("DOTNET_JitEnableOptionalRelocs", "0");
```

Comment (EgorBo) [reply src/jit-dasm-pmi/Program.cs]: @jakobbotsch renamed

**Review by jakobbotsch — APPROVED** [review]

**Comment by EgorBo** [comment]
cc @MihaZupan

**Comment by MihaZupan** [comment]
Sweet, thanks Egor

**Pull request merged**